src/ceph-disk/ceph_disk/main.py

```diff
@@ -122,7 +122,7 @@ def get_ready_by_type(what):
 
     @staticmethod
     def get_ready_by_name(name):
-        return [x[name]['ready'] for x in PTYPE.values()]
+        return [x[name]['ready'] for x in PTYPE.values() if x.get(name, None)]
```

Comment:
@tchaikov  


As my understanding, it acts as two steps:

- use get_ready_by_name to get **all UUIDs** list.
- check the given UUID is in this returned list or not?

get_read_by_name(name) is return **all UUIDs** for given name. If some PTYPE does not contains this name () (such as luks, plain do not contains lockbox) I think should skip them.  The usage of get_ready_by_name() in main.py is that after get **all UUIDs** list, it then check whether the given UUID is located in the list.

**Case 1**

```
3716         (tag, uuid) = name.split('.')
3717
3718         if tag in Ptype.get_ready_by_name('osd'):  # <- check tag in all uuids list?
3719
3720             if Ptype.is_dmcrypt(tag, 'osd'):
3721                 path = os.path.join('/dev/mapper', uuid)
3722             else:
3723                 path = os.path.join(dir, name)
```

**Case 2**

```
4075             ptype = get_partition_type(dev)
4076             LOG.debug("main_list: " + dev +
4077                       " ptype = " + str(ptype) +
4078                       " uuid = " + str(part_uuid))
4079             if ptype in Ptype.get_ready_by_name('osd'):  # check ptype in all uuids list?
4080                 if Ptype.is_dmcrypt(ptype, 'osd'):
4081                     holders = is_held(dev)
4082                     if len(holders) != 1:
```


# Example

```
PTYPE = {
    'regular': {
        ...
        'osd': {
            'ready': '4fbd7e29-9d25-41b8-afd0-062c0ceff05d',
            'tobe': '89c57f98-2fe5-4dc0-89c1-f3ad0ceff2be',
        },
        'lockbox': {
            'ready': 'fb3aabf9-d25f-47cc-bf5e-721d1816496b',
            'tobe': 'fb3aabf9-d25f-47cc-bf5e-721d181642be',
        },
    },
    'luks': {
        ...
        'osd': {
            'ready': '4fbd7e29-9d25-41b8-afd0-35865ceff05d',
            'tobe': '89c57f98-2fe5-4dc0-89c1-5ec00ceff2be',
        },
    },
    'plain': {
        ...
        'osd': {
            'ready': '4fbd7e29-9d25-41b8-afd0-5ec00ceff05d',
            'tobe': '89c57f98-2fe5-4dc0-89c1-5ec00ceff2be',
        },
    },
    'mpath': {
        ...
        'osd': {
            'ready': '4fbd7e29-8ae0-4982-bf9d-5a8d867af560',
            'tobe': '89c57f98-8ae0-4982-bf9d-5a8d867af560',
        },
        'lockbox': {
            'ready': '7f4a666a-16f3-47a2-8445-152ef4d03f6c',
            'tobe': '7f4a666a-16f3-47a2-8445-152ef4d032be',
        },
    },
}

print get_ready_by_name('osd')
print get_ready_by_name('lockbox')
```

The output should be:

```
['4fbd7e29-9d25-41b8-afd0-062c0ceff05d', '4fbd7e29-9d25-41b8-afd0-35865ceff05d', '4fbd7e29-9d25-41b8-afd0-5ec00ceff05d', '4fbd7e29-8ae0-4982-bf9d-5a8d867af560']  # <-- all osd UUIDs.
['fb3aabf9-d25f-47cc-bf5e-721d1816496b', '7f4a666a-16f3-47a2-8445-152ef4d03f6c']  # <-- all lockbox UUIDs.
```

For lockbox usage:

```
            if ptype not in ready:   # <-- Here will check whether UUID is in the list.
                LOG.warning('incorrect partition UUID: %s, expected %s'
                            % (ptype, str(ready)))
```